### PR TITLE
feat(satp-hermes): gRPC server start implementation and gateway_refactor in handlers

### PR DESCRIPTION
**Pull request by LordKubaya**
- gRPC server initialization
- usage of gateway_refactor in the handlers

### CODE REVIEW

File: packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts [hunk 1]
```diff
@@ -64,24 +79,22 @@ export class Stage1ClientHandler {
...
       throw new Error(`${fnTag}, session data is not correctly initialized`);
     }
 
-    if (!gateway.supportedDltIDs.includes(sessionData.senderGatewayNetworkId)) {
-     throw new Error(
-      `${fnTag}, recipient gateway dlt system is not supported by this gateway`,
-     );
-    }
+    // if (!gateway.supportedDltIDs.includes(sessionData.senderGatewayNetworkId)) {
+    //   throw new Error(
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts:89]: why is this code deleted?

File: packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts [hunk 2]
```diff
@@ -137,75 +156,85 @@ export class Stage1ClientHandler {
...
-    
-    await gateway.storeLog({ //todo
-      sessionID: sessionID,
-      type: "init",
-      operation: "validate",
-      data: JSON.stringify(sessionData),
-    });
+
+    // await gateway.storeLog({
+    //   //todo
```
Comment (RafaelAPB) [inline packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts:183]: why is this code deleted? we need this functionality

File: packages/cactus-plugin-satp-hermes/src/main/typescript/core/constants.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export const DEFAULT_PORT_GATEWAY_SERVER = 3010;
 export const DEFAULT_PORT_GATEWAY_CLIENT = DEFAULT_PORT_GATEWAY_SERVER + 1;
 export const DEFAULT_PORT_GATEWAY_UI = DEFAULT_PORT_GATEWAY_SERVER + 2;
+export const DEFAULT_PORT_GATEWAY_GRPC = 4000;
```
[the remark this answers is not in the document]
Comment (RafaelAPB) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/core/constants.ts]: why have you deleted this const?

File: packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts (hunk 1 above)
Comment (LordKubaya) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts:89]: This code was commented out for ease of implementing the new version, will be added back in the next PR

File: packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts (hunk 2 above)
Comment (LordKubaya) [reply packages/cactus-plugin-satp-hermes/src/main/typescript/core/stage-handlers/client/stage1-client-handler.ts:183]: Same reason from above

**Pull request merged**